templates/k8s/job-build.jinja2

```diff
@@ -0,0 +1,71 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ job_name }}
+spec:
+  backoffLimit: 3
+  template:
+    spec:
+      restartPolicy: OnFailure
+
+      # in-memory tmpfs
+      volumes:
+      - name: scratch-volume
+        emptyDir: { medium: "Memory" }
+
+      # only run on preemptible nods
+      #nodeSelector:
+      #  cloud.google.com/gke-preemptible: "true"
+
+      containers:
+      - name: kci-build
+        image: {{ "FIXME" | env_override('DOCKER_BASE') }}build-{{ "FIXME" | env_override('DOCKER_IMAGE') }}
+
+        volumeMounts:
+        - mountPath: "/scratch"
+          name: scratch-volume
```

Comment:
May be worth having a separate volume for the kernel source - again, more for building on in future than for right now. It'd be especially good if we could get to a point where we're sharing downloads between multiple builds on the same cluster, but that's definitely a separate step. I think that's working quite well for me in the experiments I'm doing. That's definitely a separate step in terms of KCI integration though.

Reply:
AFAICT it looks like we can have a shared volume between all nodes in cluster.  If that's do-able, it's probably worth rethinking the whole tarball download, and having a way for a k8s cluster to just use a local git mirror on a shared volume.

But for now, this fits well into our current tarball method.

Reply:
Yeah, we can definitely have a shared volume (though a lot of it seems annoyingly provider specific) - have to figure out if they're performant enough but hopefully. I definitely agree this is good for what we're doing with tarballs though, a lot of these were just ideas I'd had from my fiddling.